src/commands/registryCredentials/dockerLogin/AcrEnableAdminUserStep.ts

```diff
@@ -0,0 +1,55 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type ContainerRegistryManagementClient } from "@azure/arm-containerregistry";
+import { getResourceGroupFromId } from "@microsoft/vscode-azext-azureutils";
+import { activityFailIcon, activitySuccessContext, activitySuccessIcon, GenericParentTreeItem, GenericTreeItem, nonNullProp, nonNullValue } from "@microsoft/vscode-azext-utils";
+import { createActivityChildContext } from "../../../utils/activity/activityUtils";
+import { ExecuteActivityOutputStepBase, type ExecuteActivityOutput } from "../../../utils/activity/ExecuteActivityOutputStepBase";
+import { createContainerRegistryManagementClient } from "../../../utils/azureClients";
+import { localize } from "../../../utils/localize";
+import { type DockerLoginRegistryCredentialsContext } from "./DockerLoginRegistryCredentialsContext";
+
+export class AcrEnableAdminUserStep extends ExecuteActivityOutputStepBase<DockerLoginRegistryCredentialsContext> {
+    public priority: number = 450;
+
+    public async executeCore(context: DockerLoginRegistryCredentialsContext): Promise<void> {
+        const registry = nonNullValue(context.registry);
+        registry.adminUserEnabled = true;
+
+        const client: ContainerRegistryManagementClient = await createContainerRegistryManagementClient(context);
+        context.registry = await client.registries.beginUpdateAndWait(getResourceGroupFromId(nonNullProp(registry, 'id')), nonNullProp(registry, 'name'), registry);
+
+        if (!context.registry?.adminUserEnabled) {
+            throw new Error(localize('failedToUpdate', 'Failed to enable admin user for registry "{0}". Go to the portal to manually update.', context.registry?.name));
+        }
+    }
+
+    public shouldExecute(context: DockerLoginRegistryCredentialsContext): boolean {
+        return !!context.registry && !context.registry.adminUserEnabled;
```

Comment:
I feel like this shouldPrompt might need additional clarification... so:

**Case 1**: Third party registry set via registryName, resolves to false
**Case 2**: ACR set, both registryName and context.registry set, resolution determined by the 'adminUserEnabled' field
**Case 3**: ACR to be created, registryName and context.registry not set, resolves to true